pepper/apps/trade_execution.c

```diff
@@ -58,8 +58,9 @@ void compute(struct In *input, struct Out *output) {
         assert_zero(isNegative((((volume.buyVolume * order.sellAmount) - (volume.sellVolume * order.buyAmount)) * prices[fieldToInt(order.buyToken)]) - (volume.surplus * order.sellAmount)));
         totalSurplus += volume.surplus;
         
-        balances[(fieldToInt(order.account) * TOKENS) + fieldToInt(order.sellToken)] -= volume.sellVolume;
-        balances[(fieldToInt(order.account) * TOKENS) + fieldToInt(order.buyToken)] += volume.buyVolume;
+        uint32_t accountOffset = fieldToInt(order.account) * TOKENS;
```

Comment:
This is absolute 🤡🏙. If I don't pull accountOffset into it's own variable but instead use it inline as before, it calculates the offset incorrectly (order.account will be the same as order.sellToken). This seems to be a bug in pepper, where we can do one function call per line or something alike.

I will try to reduce this problem to a minimal example and raise an issue in the pepper repo.

Reply:
good catch! 
These little things are really tricky!